src/main/java/co/rsk/federate/btcreleaseclient/BtcReleaseClient.java

```diff
@@ -352,7 +352,7 @@ protected void validateTxCanBeSigned(BtcTransaction btcTx) throws FederatorAlrea
                 observedFederations.stream()
                         .forEach(f -> logger.trace("[validateTxCanBeSigned] federation p2sh redeem script {}", f.getRedeemScript()));
                 List<Federation> spendingFedFilter = observedFederations.stream()
-                        .filter(f -> (f instanceof ErpFederation ? ((ErpFederation) f).getStandardRedeemScript() : f.getRedeemScript()).equals(standardRedeemScript)).collect(Collectors.toList());
+                        .filter(f -> (f instanceof ErpFederation ? ((ErpFederation) f).getDefaultRedeemScript() : f.getRedeemScript()).equals(standardRedeemScript)).collect(Collectors.toList());
```

Comment:
```suggestion
                        .filter(f -> getRedeemScript(f).equals(standardRedeemScript)).collect(Collectors.toList());
```

What do you think about extracting the logic of getting the standardRedeemScript into a separate method? So this logic is easier to follow, putting in practice clean code recommendations :).

This could be the method, the name is just an idea, it could be better I know.
```
private Script extractStandardRedeemScript(Federation f) {
        return f instanceof ErpFederation ? ((ErpFederation) f).getDefaultRedeemScript() : f.getRedeemScript();
    }
```

Reply:
good idea! created a task for that